### DISCUSSION TITLE
Pause a single thread?

### DISCUSSION THREAD

**Discussion by jeffparsons** [ruby/debug#965] [Q&A]
Hi there! :wave: 

I've just started experimenting with using the debug gem at work. I've found it to basically work as advertised (yay!) and the VS Code integration in particular is super convenient! :sparkling_heart: 

However, my use case comes with some complications. We have some large Ruby services that hold connections to various databases, etc., and so while stepping through a single function works well, by the time I'm stepping over code that talks to the database, or if I resume execution, the whole service will have become extremely unhappy due to database servers disconnecting the service (missed heartbeats), and various other internal timeouts.

To prevent this, I'd like to be able to break only a single thread, and allow all other threads to continue. Is this possible now? Might it be possible in future, or are there restrictions in the way Ruby itself works that would prevent this? I see the readme file contains this:

> Support threads (almost done) and ractors (TODO).

But I'm not sure what that means.

Many thanks for any pointers! :bow:

**Comment by ko1** [comment]
Could you make a feature request issue?

There are two topics:

* How to implement it.
* How to design the interface.

From VSCode, DAP (Debug Adapter Protocl) has an interface so second one is solved.

BTW this kind of feature is difficult to test, though...
